src/Common/tests/System/PlatformDetection.cs

```diff
@@ -206,5 +258,73 @@ private struct RTL_OSVERSIONINFOEX
             [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 128)]
             internal string szCSDVersion;
         }
+
+        private static int s_isWindowsElevated = -1;
+
+        public static bool IsWindowsAndElevated
+        {
+            get
+            {
+                if (s_isWindowsElevated != -1)
```

Comment:
Problem is that it isn't thread safe. :( I want to create a standard tristate that we can use that is...